tests/test_execute.py

```diff
@@ -42,3 +42,47 @@ async def test_aexec(event_loop, local, code, expected_result, expected_local):
         assert stream.getvalue().strip() == repr(expected_result)
     assert local.pop("_") == expected_result
     assert local == expected_local
+
+
+# Test return and yield handling
+
+
+async def exc_from(code):
+    try:
+        await aexec(code)
+    except Exception as e:
+        return e
+    else:
+        return None
+
+
+@pytest.mark.asyncio
+async def test_return_handling():
+    error = await exc_from("return None, {}")
```

Comment:
I personally don't feel very comfortable when using a variable beyond it's "scope". I know it'll still work, but can we put `assert` into the `with` block?

Reply:
That's how it's supposed to be used though:

Reply:
That's how it's supposed to be used though: https://docs.pytest.org/en/4.6.x/reference.html#pytest-raises

> I know it'll still work, but can we put assert into the with block?

Actually no, because the exception has to be caught first.

> I personally don't feel very comfortable when using a variable beyond it's "scope".

I see what you mean, and I'd follow this line of thinking for items in a `for-loop` for instance. But AFAIK it's fine to use a context object after it's been closed to check its status (trio's [cancel scopes](https://trio.readthedocs.io/en/stable/reference-core.html?highlight=start#trio.CancelScope.cancelled_caught) is another example that comes to mind).

Reply:
Ok then 🙂